### PR TITLE
feat: custom type implements autoIncrement

### PR DESCRIPTION
**Pull request by ruanlang**
### User Case Description
ID  int64  `gorm:"column:id;type:bigint;primaryKey;autoIncrement:true" json:"id"`
https://github.com/go-gorm/gorm/issues/5505 @jinzhu @a631807682

### CODE REVIEW

**Comment by ruanlang** [comment]
@a631807682 help review

**Review by a631807682 — APPROVED** [review]
LGTM, because sequence only supports these types.
https://www.postgresql.org/docs/current/sql-createsequence.html

**Comment by jimmyduang** [comment]
@jinzhu boss,please merge

**Comment by jinzhu** [comment]
Hi @ruanlang 

Can you change it like the mysql one? 

Thank you.

**Comment by ruanlang** [comment]
@jinzhu please review

File: postgres.go [hunk 1]
```diff
@@ -192,9 +192,30 @@ func (dialector Dialector) DataTypeOf(field *schema.Field) string {
...
 		return "bytea"
+	default:
+		return dialector.getSchemaCustomType(field)
+	}
+}
+
+func (dialector Dialector) getSchemaCustomType(field *schema.Field) string {
+	sqlType := string(field.DataType)
+
+	if field.AutoIncrement {
```
Comment (jinzhu) [inline postgres.go:203]: need to check the `sqlType` doesn't contains `serial`?

Comment (ruanlang) [reply postgres.go]: ok

**Comment by jimmyduang** [comment]
@jinzhu boss，please release this pr,thx

**Pull request merged**